### ISSUE TITLE
EOF detect changes are broken

### ISSUE THREAD

**Issue by JanWielemaker** [SWI-Prolog/packages-ssl#63]
@triska: @wouterbeek reported this (note that `http://aap.nl` redirects to `https`)

```{prolog}
:- use_module(library(http/http_open)).

aap :-
	setup_call_cleanup(
	    open('aap.out', write, Out),
	    setup_call_cleanup(
		http_open('http://aap.nl', In, []),
		copy_stream_data(In, Out),
		close(In)),
	    close(Out)).
```

This signals an unexpected EOF in chunked data.   The bug disappears when reverting to
b814c2b40fe4153619d27f98076100cb622b590b

**Comment by triska** [comment]
Thank for @wouterbeek for reporting this!

I have fixed this in #65, please have a look.

**Comment by triska** [comment]
This works correctly in latest master, thank you for reporting!
